src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1679,11 +1867,154 @@ UndistributeTable(Oid relationId)
+
+
+/*
+ * AlterDistributedTable changes some properties of the given table. It uses
+ * ConvertTable function to create a new local table and move everything to that table.
+ * 
+ * The local and reference tables, tables with references, partition tables and foreign
+ * tables are not supported. The function gives errors in these cases.
+ */
+void
+AlterDistributedTable(Oid relationId, char *distributionColumn, int shardCount, char *colocateWith, bool cascadeToColocated)
+{
+	Relation relation = try_relation_open(relationId, ExclusiveLock);
+
+	if (relation == NULL)
+	{
+		ereport(ERROR, (errmsg("cannot undistribute table "
+							   "because no such distributed table exists")));
+	}
+	relation_close(relation, NoLock);
+
+	if (!IsCitusTableType(relationId, DISTRIBUTED_TABLE))
+	{
+		ereport(ERROR, (errmsg("cannot undistribute table "
+							   "because the table is not distributed")));
+	}
+
+	EnsureTableNotReferencing(relationId);
+	EnsureTableNotReferenced(relationId);
+	EnsureTableNotForeign(relationId);
+	EnsureTableNotPartition(relationId);
+	if (colocateWith != NULL && strcmp(colocateWith, "default") != 0 && strcmp(colocateWith, "none") != 0)
+	{	
+		text *colocateWithText = cstring_to_text(colocateWith);
+		Oid colocateWithTableOid = ResolveRelationId(colocateWithText, false);
+		CitusTableCacheEntry *colocateWithTableCacheEntry = GetCitusTableCacheEntry(colocateWithTableOid);
+		int colocateWithTableShardCount = colocateWithTableCacheEntry -> shardIntervalArrayLength;
+
+		if (shardCount != 0 && shardCount != colocateWithTableShardCount)
+		{
+			ereport(ERROR, (errmsg("shard_count cannot be different than the shard "
+								   "count of the table in colocate_with"),
+							errhint("if no shard_count is specified shard count "
+									"will be same with colocate_with table's")));
+		}
+
+		/*shardCount is either 0 or already same with colocateWith table's*/
+		shardCount = colocateWithTableShardCount;
+	}
+
+	ConvertTable(ALTER_DISTRIBUTED_TABLE, relationId, distributionColumn, shardCount, colocateWith, NULL, cascadeToColocated);
+}
+
+
+/*
+ * AlterTableSetAccessMethod changes the access method of the given table. It uses
+ * ConvertTable function to create a new table with the access method and move everything
+ * to that table.
+ *
+ * The local and references tables, tables with references, partition tables and foreign
+ * tables are not supported. The function gives errors in these cases.
+ */
+void
+AlterTableSetAccessMethod(Oid relationId, char *accessMethod)
+{
+	Relation relation = try_relation_open(relationId, ExclusiveLock);
+
+	if (relation == NULL)
+	{
+		ereport(ERROR, (errmsg("cannot undistribute table "
+							   "because no such distributed table exists")));
+	}
+	relation_close(relation, NoLock);
+
+	EnsureTableNotReferencing(relationId);
+	EnsureTableNotReferenced(relationId);
+	EnsureTableNotForeign(relationId);
+
+	ConvertTable(ALTER_TABLE_SET_ACCESS_METHOD, relationId, NULL, 0, NULL, accessMethod, false);
```

Comment:
it may be useful to capture the arguments in a (TableConversion) struct to keep things tidy. Potentially, if it's useful, you could even put function pointers in there.